src/formatDate.js

```diff
@@ -50,7 +50,43 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const objDate = {};
+  const currentDate = date.split(`${fromSeparator}`);
+  const toSeparator = toFormat[3];
+
+  fromFormat.forEach((format, index) => {
+    if (format !== fromSeparator) {
+      objDate[format] = currentDate[index];
+    }
+  });
+
+  toFormat.forEach((format) => {
+    if (format !== toSeparator) {
+      if (objDate[format] === undefined) {
+        objDate[format] = '';
+      }
+    }
+  });
```

Comment:
The second loop to populate objDate with empty strings for missing keys is unnecessary. Since the input is always correctly formatted, you can assume all necessary parts will be present after the first loop.